.github/workflows/main.yml

```diff
@@ -0,0 +1,68 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2.1.2
+
+      - name: Update version
+        id: update_version
+        # See https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions
+        run: |
+          $version=npm version prerelease --no-git-tag-version --preid=${{github.sha}}
+          $version=$version.substring(1)
+          echo "::set-output name=VERSION::$version"
+        shell: pwsh
+
+      - name: Install dependencies
+        run:  npm install
```

Comment:
I don't bother to user yarn.

Reply:
I think white source tool needs the yarn to install.

Reply:
I added `CodeQL` security scanning to our CI. We can add others if we want.